lib/std/wrapnils.nim

```diff
@@ -23,89 +23,98 @@ runnableExamples:
   assert ?.(f2.x2.x2).x3[] == 0
 
   assert (?.f2.x2.x2).x3 == nil  # this terminates ?. early
-  import segfaults # enable `NilAccessDefect` exceptions
+  import std/segfaults # enable `NilAccessDefect` exceptions
   doAssertRaises(NilAccessDefect): echo (?.f2.x2.x2).x3[]
 
-type Wrapnil[T] = object
-  valueImpl: T
-  validImpl: bool
+from std/options import Option, isSome, option, get, UnpackDefect
+export options.get, options.isSome, options.isNone
 
-proc wrapnil[T](a: T): Wrapnil[T] =
-  ## See top-level example.
-  Wrapnil[T](valueImpl: a, validImpl: true)
-
-template unwrap(a: Wrapnil): untyped =
-  ## See top-level example.
-  a.valueImpl
-
-template fakeDot*(a: Wrapnil, b): untyped =
+template fakeDot*(a: Option, b): untyped =
   ## See top-level example.
   let a1 = a # to avoid double evaluations
-  let a2 = a1.valueImpl
-  type T = Wrapnil[typeof(a2.b)]
-  if a1.validImpl:
+  type T = Option[typeof(a1.get.b)]
+  if isSome(a1):
+    let a2 = a1.get
```

Comment:
consider using unsafeGet?

Reply:
I don't see how that could work (please show if you have something concrete), you'd loose information on whether there was an intermediate nil in the chain, eg this must hold:
```nim
import std/wrapnils
type Foo = ref object
  x: int
var f: Foo
doAssert not (??.f.x).isSome
doAssert (??.Foo(x: 0).x).isSome
```

Reply:
?
```nim
  if isSome(a1):
    let a2 = a1.unsafeget
```

Reply:
I had misunderstood your point, you're correct! I've fixed it here and elsewhere